reloc.ml

```diff
@@ -198,27 +198,35 @@ let new_cmdline () =
   }
 
 let run_command cmdline cmd =
-  let cmd_quiet =
-    match !toolchain with
-    | `MSVC | `MSVC64 when !verbose < 1 -> cmd ^ " >NUL"
-    | _ -> cmd
+  let pipe_to_null = (!toolchain = `MSVC || !toolchain = `MSVC64) in
+  let silencer = if pipe_to_null then " >NUL" else ""
   in
   (* note: for Cygwin, using bash allow to follow symlinks to find
      gcc... *)
   if !toolchain = `CYGWIN || !toolchain = `CYGWIN64 ||
-     String.length cmd_quiet >= max_command_length
+     String.length cmd + String.length silencer > max_command_length
   then begin
     (* Dump the command in a text file and apply bash to it. *)
     let (fn, oc) = open_temp_file "longcmd" "" in
     output_string oc cmd;
+    if pipe_to_null then
+      output_string oc " >/dev/null";
     close_out oc;
 
     if !verbose >= 1 then Printf.printf "(call with bash: %s)\n%!" fn;
-    if Sys.command (Printf.sprintf "bash %s" fn) <> 0 then
+    let invoke = Printf.sprintf "bash %s" fn in
+    if Sys.command invoke <> 0 then begin
+      if pipe_to_null then begin
+        let oc = open_out fn in
+        output_string oc cmd;
+        close_out oc;
+        ignore (Sys.command invoke)
```

Comment:
What's the purpose of running the command again without the redirection?  We don't do that in the "direct" case (where bash is not used); and we might get errors (printed to stderr) twice.

Reply:
Yes we do - see deleted lines 220 and 221 below.

It's a good point, though - the treatment on the other branch is for MSVC where errors are also sent to stdout. I think the correct thing is probably to do `&>/dev/null` on this side and a belt-and-braces `2>&1` on the other?

Reply:
The whole lot exists to get rid of those pesky "foo.c" lines from MSVC...